homeassistant/components/swiss_public_transport/__init__.py

```diff
@@ -33,19 +34,25 @@ async def async_setup_entry(
     destination = config[CONF_DESTINATION]
 
     session = async_get_clientsession(hass)
-    opendata = OpendataTransport(start, destination, session)
+    opendata = OpendataTransport(start, destination, session, via=config.get(CONF_VIA))
 
     try:
         await opendata.async_get_data()
     except OpendataTransportConnectionError as e:
         raise ConfigEntryNotReady(
-            f"Timeout while connecting for entry '{start} {destination}'"
+            translation_domain=DOMAIN,
+            translation_key="request_timeout",
+            translation_placeholders={
+                "config_title": entry.title,
```

Comment:
If the error provides some details maybe also pass the actual error or write to log